packages/language-server/src/ls-config.ts

```diff
@@ -420,6 +432,93 @@ export class LSConfigManager {
     getLessConfig(): CssConfig | undefined {
         return this.lessConfig;
     }
+
+    updateTsJsFormateConfig(config: Record<TsUserConfigLang, TSUserConfig>): void {
+        (['typescript', 'javascript'] as const).forEach((lang) => {
+            if (config[lang]) {
+                this._updateTsFormatConfig(lang, config[lang]);
+            }
+        });
+        this.listeners.forEach((listener) => listener(this));
+    }
+
+    private getDefaultFormatCodeOptions(): ts.FormatCodeSettings {
+        // https://github.com/microsoft/TypeScript/blob/394f51aeed80788dca72c6f6a90d1d27886b6972/src/services/types.ts#L1014
+        return {
+            indentSize: 4,
+            tabSize: 4,
+            convertTabsToSpaces: true,
+            indentStyle: ts.IndentStyle.Smart,
+            insertSpaceAfterConstructor: false,
+            insertSpaceAfterCommaDelimiter: true,
+            insertSpaceAfterSemicolonInForStatements: true,
+            insertSpaceBeforeAndAfterBinaryOperators: true,
+            insertSpaceAfterKeywordsInControlFlowStatements: true,
+            insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis: false,
+            insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets: false,
+            insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces: true,
+            insertSpaceAfterOpeningAndBeforeClosingTemplateStringBraces: false,
+            insertSpaceAfterOpeningAndBeforeClosingJsxExpressionBraces: false,
+            insertSpaceBeforeFunctionParenthesis: false,
+            placeOpenBraceOnNewLineForFunctions: false,
+            placeOpenBraceOnNewLineForControlBlocks: false,
+            trimTrailingWhitespace: true,
+            semicolons: ts.SemicolonPreference.Ignore,
+
+            // Override TypeScript's default because VSCode default to true
+            // Also this matches the style of prettier
+            insertSpaceAfterFunctionKeywordForAnonymousFunctions: true
+        };
+    }
+
+    private _updateTsFormatConfig(lang: TsUserConfigLang, config: TSUserConfig) {
+        this.tsFormatCodeOptions[lang] = {
+            ...this.tsFormatCodeOptions[lang],
+            ...(config.format ?? {})
+        };
+    }
+
+    async getFormatCodeSettingsForFile(
+        document: Document,
+        scriptKind: ts.ScriptKind
+    ): Promise<ts.FormatCodeSettings> {
+        const filePath = document.getFilePath();
+        const configLang =
+            scriptKind === ts.ScriptKind.TS || scriptKind === ts.ScriptKind.TSX
+                ? 'typescript'
+                : 'javascript';
+
+        const tsFormatCodeOptions = this.tsFormatCodeOptions[configLang];
+
+        if (!filePath) {
+            return tsFormatCodeOptions;
+        }
+
+        const prettierConfig = this.getMergedPrettierConfig(
+            await importPrettier(filePath).resolveConfig(filePath, {
```

Comment:
I wonder if we could disable the "using prettier 2.x.x from path" logging here. It's kind of disturbing. Als it runs on completion which triggers quite often it might hurt performance.

Reply:
Yeah that makes sense